Python/sysmodule.c

```diff
@@ -2287,6 +2287,14 @@ sys_activate_stack_trampoline_impl(PyObject *module, const char *backend)
 /*[clinic end generated code: output=5783cdeb51874b43 input=a12df928758a82b4]*/
 {
 #ifdef PY_HAVE_PERF_TRAMPOLINE
+#ifdef _Py_JIT
+    _PyOptimizerObject* optimizer = _Py_GetOptimizer();
+    if (optimizer != NULL) {
+        PyErr_SetString(PyExc_ValueError, "Cannot activate the perf trampoline if the JIT is active");
```

Comment:
You're forgetting a `Py_DECREF(optimizer)` I think.

Reply:
@pablogsal I'm pretty certain there's a refleak here :(

Reply:
On it, sorry I had this on auto-land :(

Reply:
https://github.com/python/cpython/pull/126163